services/controllerhost/replicaStatsMRUCache.go

```diff
@@ -77,47 +78,51 @@ func (c *replicaStatsMRUCache) get(extent extentID, store storeID) (val *shared.
 	}
 
 	// Check for cache hit
-	if psmap, ok = c.cache[extent]; ok {
-		if val, ok = psmap.m[store]; ok { // store == `` will fall through here
-			if val != nil {
-				return
-			}
-		}
-
-		// Get any one store's replica stats; see if it is sealed
-	oneStore:
-		for foundStore, val = range psmap.m {
-			if val != nil {
-				break oneStore
-			}
-		}
+	intrfc = c.cache.Get(string(extent))
+	if intrfc == nil {
+		// Race between list ConsumerGroupExtents and list DestinationExtents
+		c.log.
+			WithField(common.TagExt, string(extent)).
+			WithField(common.TagStor, string(store)).
+			Warn(`extent cache not prepopulated`)
+		return nil
+	}
 
-		// The particular store stats that we return isn't important unless the extent has been sealed
-		// In that case, the exact count of messages available depends on which store is being read from,
-		// since the seal address will likely be different on each replica. Open extents don't have this
-		// problem, since they will only temporarily disagree on the message counts until the seal occurs
-		if val != nil && val.GetStatus() != shared.ExtentStatus_SEALED {
-			return
+	psmap = intrfc.(*perStoreReplicaStatsMap)
+	if val, ok = psmap.m[store]; ok { // store == `` will fall through here
+		if val != nil {
+			goto ttlCheck
 		}
+	}
 
-		if len(store) == 0 && val != nil {
-			return // return any found store if none specified, if applicable
+	// Get any one store's replica stats; see if it is sealed
+oneStore:
+	for foundStore, val = range psmap.m {
+		if val != nil {
+			break oneStore
 		}
 	}
 
-	// Cache miss, perform lookup and try to put
+	// The particular store stats that we return isn't important unless the extent has been sealed
+	// In that case, the exact count of messages available depends on which store is being read from,
+	// since the seal address will sometimes be different on each replica. Open extents don't have this
+	// problem, since they will only temporarily disagree on the message counts until the seal occurs
+	if val != nil && val.status == shared.ExtentStatus_OPEN {
+		goto ttlCheck
+	}
 
-	if len(store) == 0 {
-		store = foundStore //pick a random store if none specified
+	if len(store) == 0 && val != nil {
+		goto ttlCheck // return any found store if none specified, if applicable
 	}
 
+	// Cache miss, perform lookup and try to put
 	if len(store) == 0 {
-		return // This happens when a new extent comes online and there are various race conditions in Cassandra
+		store = foundStore //pick a random store if none specified
```

Comment:
don't follow - isn't this going to reset the passed arg `store` ? Why do this ? And what's the guarantee foundStore is != nil ?